### PR TITLE
AddHip in hip_net.go for modular hip

### PR DESCRIPTION
**Pull request by zycyc**
tested and compatible w/ hip.go in #270 @rcoreilly

### CODE REVIEW

**Comment by rcoreilly** [comment]
with that many args, it probably makes more sense to use a HipConfig struct as the arg to this function.  Then you can add descriptions for the meaning of each arg.   also rebase or merge to current master.

**Comment by zycyc** [comment]
should be good for review

**Review by rcoreilly — CHANGES_REQUESTED** [review]
* `In` should not be created here -- won't be used in most other models.
* projections from input(s) -> EC2 and back to conrtex etc will need to be made outside of this AddHip function -- this should just make the hip proper.
* Ec2 -> EC2 etc -- keep it consistent -- EC should always be uppercased in general.
* add `def:""` default values for all fields that have clear defaults
* Theta*, MemThr are not needed for the network config -- how about we just move those to LogConfig or a HipStatsConfig if using a common method?  Can call this HipNetConfig for better specificity.

File: axon/hip_net.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	EC3ToEC2PCon float32 `def:"0.1" desc:"percent connectivity from EC3 to EC2"`
+	EC2ToDGPCon  float32 `def:"0.25" desc:"percent connectivity from EC2 to DG"`
+	EC2ToCA3PCon float32 `def:"0.25" desc:"percent connectivity from EC2 to CA3"`
+	CA3ToCA1PCon float32 `def:"0.25" desc:"percent connectivity from CA3 to CA1"`
+	DGToCA3PCon  float32 `def:"0.02" desc:"percent connectivity into CA3 from DG"`
+
+	EC2LatRadius int     `desc:"lateral radius in EC2"`
+	EC2LatSigma  float32 `desc:"lateral sigma in EC2"`
+
+	// ECPctAct     float32 `desc:"percent activation in EC pool"`
```
Comment (rcoreilly) [inline axon/hip_net.go:33]: just cut..

**Review by rcoreilly — COMMENTED** [review]
removing those commented lines is the only change -- everything else looks good!

**Comment by rcoreilly** [comment]
also don't forget to update to master (rebase  or merge)

**Comment by zycyc** [comment]
done and rebased.

**Review by rcoreilly — APPROVED** [review]

**Pull request merged**